provider/file/file.go

```diff
@@ -18,68 +22,135 @@ var _ provider.Provider = (*Provider)(nil)
 // Provider holds configurations of the provider.
 type Provider struct {
 	provider.BaseProvider `mapstructure:",squash"`
+	Directory             string `description:"Load configuration from one or more .toml files in a directory"`
 }
 
 // Provide allows the file provider to provide configurations to traefik
 // using the given configuration channel.
 func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *safe.Pool, constraints types.Constraints) error {
-	watcher, err := fsnotify.NewWatcher()
+	var configuration *types.Configuration
+	var err error
+	var watchDir string
+
+	if p.Directory != "" {
+		watchDir = p.Directory
+		configuration, err = loadFileConfigFromDirectory(p.Directory)
+	} else {
```

Comment:
If we have both a single filename and a directory defined, the directory wins. I wonder if that's desirable or if we should have at least log a warning, if not let Traefik fail to bootstrap during a configuration sanity check. WDYT?

And whatever we decide, we should document how Traefik deals with this situation.